src/main/scala/cromwell/engine/CallExecutionActor.scala

```diff
@@ -1,42 +1,70 @@
+class CallExecutionActor() extends Actor with CromwellActor {
+  import CallExecutionActor._
+
   private val log = Logging(context.system, classOf[CallExecutionActor])
-  val tag = s"CallExecutionActor [UUID(${workflowDescriptor.shortId}):${call.name}]"
-
-  log.info(s"$tag: starting.")
-  backend.executeCommand(command, workflowDescriptor, call, callInputs, lookup) match {
-    case Success(callOutputs) =>
-      log.info(s"$tag: successful execution.")
-      context.parent ! CallActor.ExecutionFinished(callOutputs)
-    case Failure(e) =>
-      log.error(s"$tag: failed execution.")
-      context.parent ! CallActor.ExecutionFailed(e)
+  private val callExecutionActorId = UUID.randomUUID()
+
+  def tag(callReference: CallReference) = s"CallExecutionActor [UUID($callExecutionActorId)][UUID(${callReference.workflowId})/UUID(${callReference.callId})]"
+  val basicTag = s"CallExecutionActor [UUID($callExecutionActorId)]"
+
+  private def execution( workflowId: WorkflowID,
+                         backend: Backend,
+                         command: String,
+                         workflowDescriptor: WorkflowDescriptor,
+                         call: Call,
+                         callInputs: CallInputs,
+                         lookup: ScopedLookupFunction) = {
+
+    // Generate a unique CallExecution ID for this execution:
+    val callReference: CallReference = CallReference(call.callID, workflowId)
+    log.info(s"${tag(callReference)}: starting ${call.name} for workflow ${workflowDescriptor.shortId}")
+
+    val executionResult = backend.executeCommand(command, workflowDescriptor, call, callInputs, lookup, AbortFunctionRegistration(getAbortRegistrationFunction(callReference)))
+    executionResult match {
+      case Success(_) => log.info(s"${tag(callReference)}: successful execution.")
+      case Failure(e: TaskAbortedException) => log.info(s"${tag(callReference)}: aborted.")
+      case Failure(e) => log.error(s"${tag(callReference)}: failed execution - ${e.getMessage}")
+    }
+
+    context.parent ! CallActor.ExecutionFinished(callReference, call, executionResult)
   }
 
-  /**
-   * The `CallExecutionActor` is not meant to respond to messages, it just starts up, does its work,
-   * and shuts down.
-   */
   override def receive = LoggingReceive {
+    case Execute(workflowId, backend, command, workflowDescription, call, callInputs, lookup) => execution(workflowId, backend, command, workflowDescription, call, callInputs, lookup)
     case badMessage =>
-      val diagnostic = s"$tag: unexpected message $badMessage."
+      val diagnostic = s"$basicTag: unexpected message $badMessage."
       log.error(diagnostic)
   }
+  
+  private def getAbortRegistrationFunction(callReference: CallReference)(abortFunction: AbortFunction): Unit = {
+    context.parent ! CallActor.RegisterCallAbortFunction(callReference, abortFunction)
```

Comment:
The `callReference` in `RegisterCallAbortFunction` doesn't appear to be used by the receiver.